libs/fgdbg/src/Session.cpp

```diff
@@ -15,15 +15,72 @@
  */
 
 #include "fgdbg/Session.h"
+#include <sstream>
 
 namespace filament::fgdbg {
 
-void Session::addPasses(const std::vector<Pass>& passes) {}
+void Session::addPasses(const std::vector<Pass>& passes) {
+    this->passes.insert(this->passes.end(), passes.begin(), passes.end());
+}
+
+void Session::addPass(const Pass& pass) {
+    this->passes.push_back(pass);
+}
+
+void Session::addResources(const std::vector<Resource>& resources) {
+    this->resources.insert(this->resources.end(),
+            resources.begin(),
+            resources.end());
+}
 
-void Session::addResources(const std::vector<Resource>& resources) {}
+void Session::addResource(const Resource& resource) {
+    this->resources.push_back(resource);
+}
 
 void Session::update() const {
-    server->sendMessage(0);
+    // TODO(@raviola): Temporary format
+    //  <---------------passes------------------->|<----------resources----------------->
+    //  id:name:readId-readId:writeId-writeId, ...|id:name:size,id:name:size,id:name:size
```

Comment:
Have you considered using JSON or some other interchange format?

Reply:
oh... that would be cool.

Reply:
Defenitely!  Updated this logic to use YAML, let me know what you think